### DISCUSSION TITLE
Adding fstype zfs and zpool creation options

### DISCUSSION THREAD

**Discussion by Chewbakka-Wakka** [rhinstaller/anaconda#5040] [Ideas]
Much like btrfs i.e. 
part            btrfs.boot      --fstype=btrfs    --size=1024
part            btrfs.rpool     --fstype=btrfs  --size=1024

btrfs             /boot           --name=boot     --label=boot    btrfs.boot
btrfs           /                      --name=root     --label=rpool   btrfs.rpool

Syntax can be much identical:
part            zpool.boot      --fstype=zfs    --size=1024
part            zpool.rpool     --fstype=zfs  --size=1024

zpool            /boot           --name=boot     --label=boot    zpool.boot
zpool           /                      --name=root     --label=rpool   zpool.rpool

The issue is at this time... that we need zpool/zfs modules loaded installed in order to perform such actions.
Although greater adoption of ZFS on root has become possible, this is quite some process... https://openzfs.github.io/openzfs-docs/Getting%20Started/Fedora/Root%20on%20ZFS.html

**Comment by VladimirSlavik** [comment]
As the linked post mentions incompatiblities around kernel, maybe that's the first thing that should be ready before we try with the installer.

Other than that little bit ;-) First, we need support in blivet, because that's what does storage for us. Then, the rest can follow, anaconda and pykickstart in tandem. Since zpools are another kind of container, that's a bit taller order than just a new filesystem. But btrfs already is such a thing, so hopefully there's a template to follow.
